tests/ignite/contrib/metrics/test_roc_auc.py

```diff
@@ -1,117 +1,129 @@
 
-    assert roc_auc == np_roc_auc
+    test_cases = [
+        (torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10,)).long(), 1),
+        (torch.randint(0, 2, size=(100,)).long(), torch.randint(0, 2, size=(100,)).long(), 1),
+        (torch.randint(0, 2, size=(10, 1)).long(), torch.randint(0, 2, size=(10, 1)).long(), 1),
+        (torch.randint(0, 2, size=(100, 1)).long(), torch.randint(0, 2, size=(100, 1)).long(), 1),
+        # updated batches
+        (torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10,)).long(), 16),
+        (torch.randint(0, 2, size=(100,)).long(), torch.randint(0, 2, size=(100,)).long(), 16),
+        (torch.randint(0, 2, size=(10, 1)).long(), torch.randint(0, 2, size=(10, 1)).long(), 16),
+        (torch.randint(0, 2, size=(100, 1)).long(), torch.randint(0, 2, size=(100, 1)).long(), 16),
+    ]
 
+    for y_pred, y, n_iters in test_cases:
+        # check multiple random inputs as random exact occurencies are rare
+        for _ in range(10):
+            _test(y_pred, y, n_iters)
 
-def test_integration_roc_auc_score_with_activated_output_transform():
 
-    np.random.seed(1)
-    size = 100
-    np_y_pred = np.random.rand(size, 1)
-    np_y_pred_sigmoid = torch.sigmoid(torch.from_numpy(np_y_pred)).numpy()
-    np_y = np.zeros((size,), dtype=np.long)
-    np_y[size // 2 :] = 1
-    np.random.shuffle(np_y)
+def test_multilabel_input_N():
 
-    np_roc_auc = roc_auc_score(np_y, np_y_pred_sigmoid)
+    roc_auc = ROC_AUC()
 
-    batch_size = 10
+    def _test(y_pred, y, n_iters):
+        roc_auc.reset()
+        roc_auc.update((y_pred, y))
 
-    def update_fn(engine, batch):
-        idx = (engine.state.iteration - 1) * batch_size
-        y_true_batch = np_y[idx : idx + batch_size]
-        y_pred_batch = np_y_pred[idx : idx + batch_size]
-        return idx, torch.from_numpy(y_pred_batch), torch.from_numpy(y_true_batch)
+        np_y = y.numpy()
+        np_y_pred = y_pred.numpy()
 
-    engine = Engine(update_fn)
+        if n_iters > 1:
+            batch_size = y.shape[0] // n_iters + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                roc_auc.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
 
-    roc_auc_metric = ROC_AUC(output_transform=lambda x: (torch.sigmoid(x[1]), x[2]))
-    roc_auc_metric.attach(engine, "roc_auc")
+    test_cases = [
+        (torch.randint(0, 2, size=(10, 4)).long(), torch.randint(0, 2, size=(10, 4)).long(), 1),
+        (torch.randint(0, 2, size=(50, 7)).long(), torch.randint(0, 2, size=(50, 7)).long(), 1),
+        (torch.randint(0, 2, size=(100, 4)).long(), torch.randint(0, 2, size=(100, 4)).long(), 1),
+        (torch.randint(0, 2, size=(200, 6)).long(), torch.randint(0, 2, size=(200, 6)).long(), 1),
+        # updated batches
+        (torch.randint(0, 2, size=(10, 4)).long(), torch.randint(0, 2, size=(10, 4)).long(), 16),
```

Comment:
I think this test does not make sense as `batch_size = y.shape[0] // n_iters + 1 = 1`.

Reply:
That's right! that's how it should be:
```
for y_pred, y, batch_size in test_cases:
    ......

if batch_size > 1:
            n_iters = y.shape[0] // batch_size + 1
```
I think it should be updated now in `average_precision` and other tests, right?

Reply:
Yes, something like that.
And yes, we should fix other as well if it is the same case. Even it would be better to code such test function once and reuse it instead of copying multiple times...